### PR TITLE
Allow brace completion in comments

### PR DESCRIPTION
**Pull request by anubmat**
Allows brace completion inside comments in JS and TS files

### CODE REVIEW

**Comment by anubmat** [comment]
@jramsay , @paulvanbrenk , @billti , @vlad, Can someone please review?

File: src/services/services.ts [hunk 1]
```diff
@@ -1725,7 +1725,7 @@ namespace ts {
             const sourceFile = syntaxTreeCache.getCurrentSourceFile(fileName);
 
             // Check if in a context where we don't want to perform any insertion
-            if (isInString(sourceFile, position) || isInComment(sourceFile, position)) {
```
Comment (paulvanbrenk) [inline src/services/services.ts]: This should only be the case for `{` not for all braces. IMO

Comment (anubmat) [inline src/services/services.ts]: c# is smart, otherwise too outside comments, it only completes when brackets are placed in an expected location

Comment (paulvanbrenk) [inline src/services/services.ts]: Ok, Don't have a strong opinion either way..

**Comment by billti** [comment]
How does this behave in the editor? If we auto-complete the closing brace, (so after typing the opening brace the cursor is at `( | )` ), then when I type a closing brace, does it automatically just move me past the auto-inserted one, and not add an additional closing brace (i.e. I should end up with `( ) |`, not `( ) | )` ).

**Comment by anubmat** [comment]
The behavior is same as outside comments, i.e. on typing the closing brace, it moves past the auto-inserted one and does not add an additional closing brace, like that ```
( ) | ```

**Comment by billti** [comment]
Great, thanks. If you've tried it out in VS and VS Code and the experience feels right, then 👍  :-)

**Comment by anubmat** [comment]
I tried and it works nicely in VS. As for VsCode, brace completion was already working inside comments.

**Pull request merged**